source/common/http/header_map_impl.cc

```diff
@@ -88,7 +93,9 @@ void HeaderString::append(const char* data, uint32_t size) {
     type_ = Type::Dynamic;
     dynamic_capacity_ =
         std::max(MinDynamicCapacity, static_cast<size_t>((string_length_ + size) * 2));
-    validateCapacity(dynamic_capacity_);
+    if (dynamic_capacity_ != MinDynamicCapacity) {
+      validateCapacity(newCapacity(string_length_, size));
```

Comment:
This seems scary to me. Can't assign to a `size_t` above the target new capacity, validate that specific value and then allocate with the same value? Seems hard to reconcile this `if` block with what is actually allocated.

Reply:
Actually it's also bad because if we overflow to exactly MinDynamicCapacity it's broken.  Overflow is hard